econml/ortho_forest.py

```diff
@@ -330,7 +374,26 @@ def _get_weights(self, X_single):
                 w2[unique] += leaf_weight * counts
             else:
                 w2[weight_indexes] += leaf_weight
-        return (w1 / self.n_trees, w2 / self.n_trees)
+        return (w1 / len(tree_range), w2 / len(tree_range))
+
+    def _get_blb_indices(self, X):
+        """Get  data indices for every tree under the little bags split."""
+        # Define subsample size
+        subsample_size = X.shape[0] // 2
+        if not self.bootstrap:
+            if self.subsample_ratio > 1.0:
+                # Safety check
+                self.subsample_ratio = 1.0
+            subsample_size = int(self.subsample_ratio * subsample_size)
+        subsample_ind = []
+        # Draw points to create little bags
+        for it in range(self.n_slices):
+            half_sample_inds = self.random_state.choice(
+                X.shape[0], X.shape[0] // 2, replace=False)
+            for _ in np.arange(it * self.slice_len, min((it + 1) * self.slice_len, self.n_trees)):
+                subsample_ind.append(half_sample_inds[self.random_state.choice(
+                    X.shape[0] // 2, subsample_size, replace=False)])
```

Comment:
Is this right - it's just a permutation?

```suggestion
                    subsample_size, subsample_size, replace=False)])
```

Reply:
If so, consider using `permutation` directly instead of `choice` to make the intent clear.

Reply:
No, because `subsample_size` gets updated if bootstrap=False (which is the default): `subsample_size = int(self.subsample_ratio * subsample_size)`